server/chess.py

```diff
@@ -256,7 +256,7 @@ def validate_king_move(
         """Validate a king's move."""
         absolute_file_delta = abs(file - king.file)
         absolute_rank_delta = abs(rank - king.rank)
-        if (not absolute_rank_delta) and not king.has_moved:
+        if (not absolute_rank_delta) and not king.has_moved and absolute_file_delta > 1:
```

Comment:
This extra check is unnecessary, The target file is checked immediately after.

Reply:
It is not unnecessary and you have missinterpreted the first line about castling. The way you had it programmed meant that the ONLY move a king could make as his first move was a castle.  However the king can make loads of other moves as his first move instead.
This check means that the following validation routine is only performed on castling moves, all other moves are still valid and are dealt with after the if statement.

Reply:
Uh... This is too much for me this late, will take another look tomorrow